### ISSUE TITLE
DKIM Failing on multipart messages

### ISSUE THREAD

**Issue by deepvoice** [zone-eu/zone-mta#99]
dkim-relaxed-body.js 

If the message is *not* a multipart mime it creates the body hash correctly.
If it is multipart the body hash is not generating correctly and is failing DKIM checks

**Comment by andris9** [comment]
ZoneMTA signs multipart messages just fine. Are you using the latest version of ZoneMTA? Can you provide an example message that fails signing?

**Comment by deepvoice** [comment]
using [smtp-cli](http://www.logix.cz/michal/devel/smtp-cli/) I am able recreate the issue no matter what files I am using.

smtp-cli  --from user@domain --to user@example.com --subject "Multi-part test" --body-html /path/to/file.html --body-text /path/to/file.txt --host 127.0.0.1

**Comment by andris9** [comment]
I'm not able to produce a message with an invalid signature myself. Do you have an example that can be verified?

**Comment by rquast** [comment]
I'm also experiencing dkim signing issues that may be related to this. Sorry, I had a ticket open a while ago for it but never got around to trying to reproduce this. Using the latest code from repo with a slightly modified config and my keys for signing.

**Comment by rquast** [comment]
Here is what I post to the zone-api API:

https://gist.github.com/rquast/006235e5358369fc39b744e1e826f77b

Here is the PHP/Curl code I use to send the message:

https://gist.github.com/rquast/434faef84db143a7d2d9f664c4b5b375

I suspect it has something to do with the encoding of html. The message is turning into "quoted-printable" instead of 7bit.

**Comment by rquast** [comment]
I took that json and modified the html right down to a little bit, then resent and DKIM encoding worked.

```
----_NmP-2c2a52d4c8615e18-Part_1
Content-Type: text/html
Content-Transfer-Encoding: 7bit

<html><head></head><body>Hello World</body></html>
----_NmP-2c2a52d4c8615e18-Part_1--
```

That time it used 7bit encoding, and the body hash was verified. There's something funky going on with the html encoding.

**Comment by rquast** [comment]
Found it! If you put a \n in html, it will break. :\

**Comment by rquast** [comment]
I notice your test has this code. This is REALLY important, or the DKIM will fail.

https://github.com/zone-eu/zone-mta/blob/master/test/dkim-relaxed-body-test.js#L10

**Comment by andris9** [comment]
@rquast Oh, I was assuming you are sending using SMTP and not API. Using API did not properly handle the newlines indeed, I fixed it and released zone-mta@1.3.1
